### PR TITLE
Transformation schema for nodes and edges with view centric and model centric

### PR DESCRIPTION
**Pull request by nimeshawij**
## Description
Transformation schema for nodes, edges in both view centric and model centric

  \_api\transformations\schema.py

1. Added condition to check destination.type "nodes" or "edges" or "instances", then set the resource path /transformation/schema/instances. 

Because all other resource paths are in the format /transformation/schema/{destination.type}

2.  `withInstanceSpace`, `isConnectionDefinition`, `instanceType` are optional 

3.  Examples

- Get the schema for a transformation producing nodes::
- Get the schema for a transformation producing edges::
- Get the schema for a transformation producing instance type data model::
-  Get the schema for a transformation producing instance relationship data model::

tests\tests_integration\test_api\test_transformations\test_schema.py

Added
test_nodes_with_view(self, cognite_client)
test_edges_with_view(self, cognite_client)
test_instance_type_data_model(self, cognite_client)
test_instance_relationship_data_model(self, cognite_client)

## Checklist:
- [x] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/_api/transformations/schema.py
Comment (audunska) [inline cognite/client/_api/transformations/schema.py:23]: Should probably add documentation for these under `Args`: below?

File: cognite/client/_api/transformations/schema.py
Comment (audunska) [inline cognite/client/_api/transformations/schema.py:77]: Might have an example using the new parameters?

**Review by audunska — APPROVED** [review]
Looks mostly good. I left some suggestions.

File: cognite/client/_api/transformations/schema.py
Comment (nimeshawij) [reply cognite/client/_api/transformations/schema.py:77]: In the user document https://cognite-sdk-python.readthedocs-hosted.com/en/latest/transformations.html#create-transformations, there are no examples with real data for the Transformation `create`. 

If I added real data examples, then the format will break. Please share you opinion.

**Comment by nimeshawij** [comment]
@audunska This PR cannot merge. Do I need more reviewers?

**Comment by nimeshawij** [comment]
@SunilVenekarKrishnamoorthy waiting for + one person approval to merge.

File: cognite/client/_api/transformations/schema.py
Comment (erlendvollset) [inline cognite/client/_api/transformations/schema.py:31]: What is this parameter? It does not make sense to me, even with the explanation.

File: cognite/client/_api/transformations/schema.py
Comment (erlendvollset) [inline cognite/client/_api/transformations/schema.py:23]: These parameters are only applicable for the `Instances` TransformationDestination? 

Then we should either add overloads to reflect that, or create a separate method.

Comment (erlendvollset) [reply cognite/client/_api/transformations/schema.py:23]: separate method is probably easier.

Comment (nimeshawij) [reply cognite/client/_api/transformations/schema.py:23]: Yes, applicable for `Instances `TransformationDestination. However, if I defined in separate method, then the request method will be different. 

Currently all the destination use the method; so that's the reason I used condition to check the 'Instances', 'nodes', and 'edges'

`TransformationSchemaAPI.retrieve(destination: cognite.client.data_classes.transformations.common.TransformationDestination, conflict_mode: Optional[str] = None) → cognite.client.data_classes.transformations.schema.TransformationSchemaColumnList`

Comment (erlendvollset) [reply cognite/client/_api/transformations/schema.py:23]: > if I defined in separate method, then the request method will be different.
Not following. Could you elaborate?

Let's just introduce a `retrieve_instances_schema` method

File: cognite/client/_api/transformations/schema.py
Comment (erlendvollset) [inline cognite/client/_api/transformations/schema.py:89]: unused

File: cognite/client/_api/transformations/schema.py
Comment (erlendvollset) [inline cognite/client/_api/transformations/schema.py:112]: unused

File: cognite/client/_api/transformations/schema.py
Comment (erlendvollset) [inline cognite/client/_api/transformations/schema.py:121]: This is identical for both branches of the conditional - can be moved out.

File: cognite/client/_api/transformations/schema.py
Comment (nimeshawij) [reply cognite/client/_api/transformations/schema.py:23]: All the destination types are use the `transformations.schema.retrieve` method; so defining separate method like for example  `transformations.schema.retrieve_instances` will not be a good opinion according to my understanding.  

I can try with overload option by providing multiple versions of the `retrieve `method;

```
def retrieve(self, destination: InstancesTransformationDestination) -> TransformationSchemaColumnList:
    # Implementation for Instances transformation destination
    pass

def retrieve(self, destination: TransformationDestination, conflict_mode: str) -> TransformationSchemaColumnList:
    # Implementation for other transformation destinations
    pass
```

File: cognite/client/_api/transformations/schema.py
Comment (nimeshawij) [reply cognite/client/_api/transformations/schema.py]: Resolved.

Line 105

```
 def _retrieve(
        self, destination: TransformationDestination, other_params: Optional[dict]
    ) -> TransformationSchemaColumnList:
```

File: cognite/client/_api/transformations/schema.py
Comment (nimeshawij) [reply cognite/client/_api/transformations/schema.py]: Added condition to defined value for `url_path` for instances and all the other schemas.  

```
if destination.type in ["nodes", "edges", "instances"]:
            url_path = utils._auxiliary.interpolate_and_url_encode(self._RESOURCE_PATH + "/{}", "instances")
        else:
            url_path = utils._auxiliary.interpolate_and_url_encode(self._RESOURCE_PATH + "/{}", str(destination.type))
```

https://developer.cognite.com/api/v1/#tag/Schema/operation/getFlexibleDataModelInstanceSchema

File: cognite/client/_api/transformations/schema.py
Comment (nimeshawij) [reply cognite/client/_api/transformations/schema.py:28]: withInstanceSpace - # optional, if specified it takes  precedence over the space mentionled in the sql query

 isConnectionDefinition -  if you pass the query `isConnectionDefinition=true`, we won’t send back the type as part of the schema

File: cognite/client/_api/transformations/schema.py
Comment (nimeshawij) [reply cognite/client/_api/transformations/schema.py]: Unfortunately, In Python, method overloading is not supported in the same way as in statically typed languages.

I got below error in git workflow; however, the code is run local machine without any error. 

```
cognite/client/_api/transformations/schema.py:87: error: Name "retrieve" already defined on line 17  [no-redef]
        def retrieve(
        ^
```

If we defined new method to `retrieve` transformation schemas endpoint for Instance destination only (ex: `TransformationSchemaAPI.retrieve_instances)` then inconsistence break. Document https://cognite-sdk-python.readthedocs-hosted.com/en/latest/transformations.html#get-transformation-schema;

My take on this, the request `TransformationSchemaAPI.retrieve() `should have for all the schema endpoints. 

So, the only way is to define `retrieve `method as shown below;

```
def retrieve(
    self,
    destination: TransformationDestination,
    conflict_mode: Optional[str] = None,
    with_instance_space: Optional[bool] = None,
    is_connection_definition: Optional[bool] = None,
    instance_type: Optional[str] = None,
) -> TransformationSchemaColumnList:
```

**Comment by nimeshawij** [comment]
@erlendvollset Requested changers has been done. Could you please review the code? 

`build/test_full `failed due to below error, which is not related to this PR. 

```
    ]
FAILED tests/tests_integration/test_api/test_data_modeling/test_instances.py::TestInstancesAPI::test_retrieve_nodes_and_edges - AssertionError: assert [NodeId(space...keitel'), ...] == [NodeId(space...keitel'), ...]
  At index 58 diff: NodeId(space='erlendtestspace', external_id='test2') != NodeId(space='erlendtestspace', external_id='test')
  Full diff:
```

**Comment by haakonvt** [comment]
Closing as stale. Feel free to reopen when you have time to work on this!

**Pull request closed without merging**
